common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -2284,54 +2292,51 @@ func (d *cassandraPersistence) LeaseTaskQueue(request *p.LeaseTaskQueueRequest)
 func (d *cassandraPersistence) UpdateTaskQueue(request *p.UpdateTaskQueueRequest) (*p.UpdateTaskQueueResponse, error) {
 	tli := *request.TaskQueueInfo
 	tli.LastUpdateTime = timestamp.TimeNowPtrUtc()
-	if tli.Kind == enumspb.TASK_QUEUE_KIND_STICKY { // if task_queue is sticky, then update with TTL
-		expiry := types.TimestampNow()
-		expiry.Seconds += int64(stickyTaskQueueTTL)
-
-		datablob, err := serialization.TaskQueueInfoToBlob(&tli)
-		if err != nil {
-			return nil, convertCommonErrors("UpdateTaskQueue", err)
-		}
+	datablob, err := serialization.TaskQueueInfoToBlob(&tli)
```

Comment:
In the MySQL Persistence layer, we actually set the ExpiryTime field on the TLI object before serializing it. What's the reason for the difference between the two layers?

Reply:
Cassandra has TTL support, SQL not.

There should be a worker / background logic scan SQL to get rid of stale records